vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/model/descriptor/struct/StructDescriptor.kt

```diff
@@ -15,4 +15,4 @@ import org.vitrivr.engine.core.model.descriptor.Descriptor
  * @author Ralph Gasser
  * @version 1.0.0
  */
-interface StructDescriptor : Descriptor
+interface StructDescriptor<T : StructDescriptor<T>> : Descriptor<T>
```

Comment:
Do we have any `StructDescriptor`s that are not `MapStructDescriptor`s or do we foresee any need to have them in the future? If no, why do we even have the distinction?

Reply:
We currently don't. I've migrated all of them. It would, however, be possible in theory.

From a persistence perspective, `MapStructDescriptor`s are very convenient. I currently cannot foresee a use-case where one would use something else. Hence, yes, we could desig all `StructDescriptor`s to be `MapStructDescriptor`s.

Reply:
Okay so I removed the old interface. `StructDescriptor` is now an abstract class basically implementing the former `MapStructDescriptor` logic.